src/09-coveio_abi.adoc

```diff
@@ -686,3 +686,216 @@ The possible error codes returned in `sbiret.error` are shown below.
 | SBI_ERR_ALREADY_STOPPED | The device interface is already stopped.
 | SBI_ERR_FAILED          | The operation failed for unknown reasons.
 |===
+
+=== CoVE IO TSM Extension (EID #0x434F5654 "COVT")
+
+==== PCIe Root Port IDE Management
+
+[#sbi_covt_program_rp_ide_key]
+===== Function: CoVE TSM Program PCIe Root Port IDE Key (FID #1024)
+[source, C]
+-----
+struct sbiret sbi_covt_program_rp_ide_key(unsigned long rp_id,
+                                          unsigned long key_desc_addr,
+                                          unsigned long key_data_addr);
+-----
+
+Programs a PCIe IDE key into the `rp_id` PCIe Root Port. This is functionally
+equivalent to the PCIe IDE KM `KEY_PROG` command.
+
+The IDE key descriptor at `key_desc_addr` must be structured as follows:
```

Comment:
key_desc is only 4 bytes. Do we really need an `addr` to point to a 4 bytes structure?

Or, we can use a UINT32 (4 bytes) for key_desc.